src/main/java/com/woowacamp/soolsool/core/member/service/MemberService.java

```diff
@@ -21,26 +30,49 @@ public class MemberService {
 
     private final MemberRepository memberRepository;
     private final MemberRoleRepository memberRoleRepository;
+    private final MemberMileageChargeRepository memberMileageChargeRepository;
 
     @Transactional
     public void addMember(final MemberAddRequest memberAddRequest) {
-        final MemberRole memberRole = memberRoleRepository.findById(1L)
-            .orElseThrow(() -> new SoolSoolException(MemberErrorCode.MEMBER_NO_ROLE_TYPE));
+        checkDuplicatedEmail(memberAddRequest.getEmail());
+
+        final MemberRole memberRole = getMemberRole(memberAddRequest);
         final Member member = memberAddRequest.toMember(memberRole);
         memberRepository.save(member);
     }
 
+    private void checkDuplicatedEmail(final String email) {
+        final Optional<Member> duplicatedEmil = memberRepository
+            .findByEmail(new MemberEmail(email));
+
+        if (duplicatedEmil.isPresent()) {
+            throw new SoolSoolException(MemberErrorCode.MEMBER_DUPLICATED_EMAIL);
+        }
+    }
```

Comment:
```suggestion
    private void checkDuplicatedEmail(final String email) {
        if (memberRepository.findByEmail(new MemberEmail(email)).isPresent()) {
            throw new SoolSoolException(MemberErrorCode.MEMBER_DUPLICATED_EMAIL);
        }
    }
```

지역 변수의 범위를 최소화하는게 좋아 보여요

Reply:
원래는 아래처럼 한줄로 했는데 가독성 측면에서 이해하기 어려울것 같아서 일부로 분리했는데 어떻게 생각해?

Reply:
지역 변수를 불필요하게 밖으로 꺼내면 다른 데서 사용될 것처럼 오해를 살 수 있다고 하더라고요. 취향 리뷰입니다!